### ISSUE TITLE
Runtime error 015 at 1DC3:06A7 when launching TDL on MiSTer

### ISSUE THREAD

**Issue by goldenegg** [Voljega/ExoDOSConverter#90]
I've been trying to create my own game VHD for MiSTer using ExoDOSConverter 0.9.2-beta. TDL can't load on any of the VHDs I've created. They come back with "Runtime error 015 at 1DC3:06A7" when trying to launch TDL.

Also, when using the games VHD with a boot VHD from flynnsbit, it always tries to read from the CD whenever performing actions on the D:. Not sure if that's an issue with the boot VHD or the one I created with the converter tool.

**Comment by Voljega** [comment]
I'm away on vacation at the moment and it might be more @flynnsbit's domain
than mine, he might be able to help you on both these issues

Le lun. 26 juil. 2021 à 17:03, goldenegg ***@***.***> a
écrit :

**Comment by goldenegg** [comment]
Thanks.

@flynnsbit - If it would be helpful, I can throw a VHD up on Mega or somewhere else if you want to take a look at how it's built.

**Comment by flynnsbit** [comment]
runtime error is usually your tdl.ini isn't in place, or the handlers.ini file, or a reference point in them is referencing nothing. Something like that.  If the index got corrupt that could do it.  If you attach it here I can take a look.  tdl.exe, index*.* handlers*

**Comment by goldenegg** [comment]
[TDL_FILES.zip](https://github.com/Voljega/ExoDOSConverter/files/6890046/TDL_FILES.zip)

I've attached the TDL files here.

I've got the VHDs you've created. The boot VHDs act one of two ways with my VHD files. They either provide proper access to D:, resulting in the runtime error, or they reassign D: to F: so it's constantly trying to read from the empty CD-ROM whenever D: is accessed.

**Comment by flynnsbit** [comment]
1. when setting up your data VHD that hosts the TDL exe and games, are you creating two partitions in the VHD?  The stub disk for D: that is then replaced with the CD-Drive drive letter, and the E: which is actually where all the files are?  
2.  You can also get a runtime error if you have a reference in HANDLERS.INI where that EXE/program and it doesn't actually exist.  I "Think" I included dump.exe for the SYX extension, and the pctview.bat under utils\pictview for the JPG extension.viewer.

**Comment by goldenegg** [comment]
I wasn't aware of the dummy volume. I'll create a new VHD with one, as that should definitely solve the problem with the most recent boot VHD.

I checked HANDLERS.INI and all files referenced in there exist and are in the correct subdirectories. I'll see if the new VHD, with the dummy volume, has any impact here.

**Comment by goldenegg** [comment]
Got things working. I had to make two changes.

1. I created the dummy volume, which fixed the problem of the CD-ROM mapping
2. When copying files to the VHD, for some reason doing a drag and drop always causes the runtime error. If I use XCOPY, everything works. Not quite sure what's happening with Windows file copy, but this might be useful for other people who might have trouble with their VHDs.

Thanks for the help!
